### ISSUE TITLE
Not possible to run two instances of appsync-emulator at once on different ports

### ISSUE THREAD

**Issue by mattgibson** [ConduitVC/aws-utils#115]
I'm running the serverless appsync emulator library via serverless appsync offline. I want to be able to have two instances running - one for development and one for locally running browser specs. These need to be on different ports, but when I do this, two instances of `mosca` are started for the graphql subscriptions and they both use port 3000 on loclahost so they conflict.

I'm not sure if the right solution is to have the serverless emulator choose a random port or to have mosca do it.

```
serverless appsync-offline start --port 37000
serverless appsync-offline start --port 47000

{"pid":43621,"hostname":"Matts MacBook Pro","name":"mosca","level":50,"time":1556740318814,"msg":"Error starting Mosca Server","v":1}
Serverless: ERROR: Error: listen EADDRINUSE: address already in use :::1883
```

**Comment by cbaron** [comment]
Hi @mattgibson -- this is a known issue.

I think somewhere near [here](https://github.com/ConduitVC/aws-utils/blob/22f649c73270bd95718e2951ccb763863f4e1af9/packages/appsync-emulator-serverless/serverCore.js#L321) we need to tell mosca to choose an unused port.

We will investigate and get back to you.  Let us know if you come up with anything.

**Comment by lightsofapollo** [comment]
Hrm I re-read that code and looked up the port `1883` and now know it has something to do with the mqtt setup.

**Comment by lightsofapollo** [comment]
Previous attempt at fixing this issue was : https://github.com/ConduitVC/aws-utils/commit/99bb347c494296fcc399b9735437158a8b36cebd

**Comment by mattgibson** [comment]
Looking at where mosca is started here https://github.com/ConduitVC/aws-utils/blob/master/packages/appsync-emulator-serverless/serverCore.js#L323

I found I could fix the error in one line if I did this:

```
  const mqttServer = new mosca.Server({
    port: pareseInt(port, 10) + 1,
    backend: { type: 'memory' },
  });
```

But i'm not sure if that actually broke the subscriptions or not. it just fixed the error and let me start both instances.

**Comment by lightsofapollo** [comment]
My hypothesis is there is an *additional* port bound by mosca that we are in fact not using. There should be some mechanism to turn that off (I am investigating now)

**Comment by lightsofapollo** [comment]
I can reproduce this issue on `^0.7.1` (in our test repo) but cannot reproduce on master...

**Comment by lightsofapollo** [comment]
I found another issue on master and will publish a version shortly which will include this fix ! I also updated our test repo to run tests in parallel to verify we don't break this in future.

**Comment by lightsofapollo** [comment]
Please try `0.12` and see if that works better! I published a few fixes.

**Comment by lightsofapollo** [comment]
Marking this as closed . please reopen if issue persists.

**Comment by mattgibson** [comment]
@lightsofapollo Thanks for the quick fix! Version `0.12` does fix the issue of not being able to run two instances, but I was unable to verify that the subscriptions function correctly. I'll open another issue if that turns out to be the case in future.
